File: src/main/java/com/market/jw/Book.java
package com.market.jw;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Getter
@ToString
@Builder
@AllArgsConstructor(access = AccessLevel.PROTECTED)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "Books")
public class Book extends BulkInsertEntity {

    @Id
    @Column
    private Long bookSeq;

    @Column
    private String title;

    @Column
    private String author;

    @Column
    private LocalDateTime publicationDate;

    @Column(precision = 20, scale = 5)
    private BigDecimal price;

    @Override
    public Long getId() {
        return bookSeq;
    }

    public static Book from(BookSaveDto dto) {
        return Book.builder()
                .bookSeq(UUID.randomUUID().getMostSignificantBits())
                .title(dto.getTitle())
                .author(dto.getAuthor())
                .publicationDate(dto.getPublicationDate())
                .price(dto.getPrice())
                .build();
    }
}

File: src/main/kotlin/com/example/bookstore/domain/book/entity/Book.kt
package com.example.bookstore.domain.book.entity

import com.example.bookstore.common.jpa.entity.BulkInsertEntityJ
import java.math.BigDecimal
import java.time.LocalDateTime
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "Books")
class Book private constructor(
    bookSeq: Long,
    title: String,
    author: String,
    publicationDate: LocalDateTime? = null,
    price: BigDecimal? = null
) : BulkInsertEntityJ() {

    @Id
    @Column
    var bookSeq: Long = bookSeq
        protected set

    @Column
    var title: String = title
        protected set

    @Column
    var author: String = author
        protected set

    @Column
    var publicationDate: LocalDateTime? = publicationDate
        protected set

    @Column
    var price: BigDecimal? = price
        protected set

    override fun getId(): Long {
        return bookSeq
    }

    fun updatePrice(price: BigDecimal) {
        this.price = price
    }

    companion object {
        fun create(title: String, author: String, price: BigDecimal) {
            
        }
    }

}
